src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamAppealDecidedPersonalisation.java

```diff
@@ -0,0 +1,117 @@
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.RequiredFieldMissingException;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AppealDecision;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCaseDefinition;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.personalisation.EmailWithLinkNotificationPersonalisation;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.service.DetEmailService;
+import uk.gov.hmcts.reform.iacasenotificationsapi.infrastructure.clients.DocumentDownloadClient;
+import uk.gov.service.notify.NotificationClientException;
+
+@Slf4j
+@Service
+public class DetentionEngagementTeamAppealDecidedPersonalisation implements EmailWithLinkNotificationPersonalisation {
+
+    private final String detentionEngagementTeamUploadAppealDecidedDismissedTemplateId;
+    private final String detentionEngagementTeamUploadAppealDecidedAllowedTemplateId;
+    private final DetEmailService detEmailService;
+    private final DocumentDownloadClient documentDownloadClient;
+
+    @Value("${govnotify.emailPrefix.adaInPerson}")
+    private String adaSubjectPrefix;
+
+    @Value("${govnotify.emailPrefix.nonAdaInPerson}")
+    private String iaftSubjectPrefix;
+
+    public DetentionEngagementTeamAppealDecidedPersonalisation(
+            @Value("${govnotify.template.appealDecided.detentionTeam.dismissed.email}") String detentionEngagementTeamUploadAppealDecidedDismissedTemplateId,
+            @Value("${govnotify.template.appealDecided.detentionTeam.allowed.email}") String detentionEngagementTeamUploadAppealDecidedAllowedTemplateId,
+            DetEmailService detEmailService,
+            DocumentDownloadClient documentDownloadClient
+    ) {
+        this.detentionEngagementTeamUploadAppealDecidedDismissedTemplateId = detentionEngagementTeamUploadAppealDecidedDismissedTemplateId;
+        this.detentionEngagementTeamUploadAppealDecidedAllowedTemplateId = detentionEngagementTeamUploadAppealDecidedAllowedTemplateId;
+        this.detEmailService = detEmailService;
+        this.documentDownloadClient = documentDownloadClient;
+    }
+
+    @Override
+    public String getReferenceId(Long caseId) {
+        return caseId + "_INTERNAL_ADA_APPEAL_DECIDED_EMAIL";
+    }
+
+    @Override
+    public String getTemplateId(AsylumCase asylumCase) {
+        return getAppealDecisionFrom(asylumCase).equals(AppealDecision.DISMISSED)
+                ? detentionEngagementTeamUploadAppealDecidedDismissedTemplateId : detentionEngagementTeamUploadAppealDecidedAllowedTemplateId;
+    }
+
+    @Override
+    public Set<String> getRecipientsList(AsylumCase asylumCase) {
+        return Collections.singleton(detEmailService.getDetEmailAddress(asylumCase));
```

Comment:
We also put a check here for the detention facility. If it's empty or 'other' then return an emptySet so no notification is sent, otherwise return the email from detEmailService.

Reply:
Added a check for detention facillity.